File: collect_app/src/androidTest/java/org/odk/collect/android/support/RunnableRule.java
package org.odk.collect.android.support;

import org.junit.rules.TestRule;
import org.junit.runner.Description;
import org.junit.runners.model.Statement;

public class RunnableRule implements TestRule {

    private final Runnable runnable;

    public RunnableRule(Runnable runnable) {
        this.runnable = runnable;
    }

    @Override
    public Statement apply(Statement base, Description description) {
        return new Statement() {
            @Override
            public void evaluate() throws Throwable {
                runnable.run();
                base.evaluate();
            }
        };
    }
}

File: collect_app/src/androidTest/java/org/odk/collect/android/support/rules/RunnableRule.kt
package org.odk.collect.android.support.rules

import org.junit.rules.TestRule
import org.junit.runner.Description
import org.junit.runners.model.Statement

class RunnableRule(
    private val runnable: Runnable
) : TestRule {

    override fun apply(base: Statement, description: Description): Statement =
        object : Statement() {
            override fun evaluate() {
                runnable.run()
                base.evaluate()
            }
        }
}
